### ISSUE TITLE
Table package_license_info_from_files has no primary key

### ISSUE THREAD

**Issue by ttgurney** [socs-dev-env/Database-SPDX2.0#1]
```
CREATE TABLE IF NOT EXISTS `package_license_info_from_files` (
  `package_id` INT NOT NULL,
  `package_license_info_from_files` varchar(255) NOT NULL,
  `created_at` datetime NOT NULL,
  `updated_at` datetime DEFAULT NULL,
  FOREIGN KEY (`package_id`) REFERENCES packages(`id`)
);
```

**Comment by zwmcfarland** [comment]
Do you have a need for a primary key?

**Comment by ttgurney** [comment]
Yes. I'm using an ORM which is expecting this table to have a primary key and won't create a mapping for it without one.

But it's beside the point, IMO (I wouldn't expect a table's schema to change just because an ORM complains about it) because having a primary key on every table is a best practice. Except for rare circumstances (e.g. log tables), I would consider a missing primary key to be a bug.

**Comment by zwmcfarland** [comment]
Sorry I was thinking identity, i will have to check if we can set the package id as the primary or if there will be multiple. I'll update when I get to a computer.

**Comment by ttgurney** [comment]
Actually, this raises an important question. Does this need its own separate table or is it a one-to-one relationship that can be included in the `packages` table?

My understanding of this table is that it stores a list of licenses found in the files in a package (section 3.13 in the [SPDX 2.0 spec](http://spdx.org/sites/spdx/files/SPDX-2.0.pdf)). So would this make better sense as a many-to-many? (for example a `packages` table, `licenses` table and a `packages_licenses` table)

**Comment by zwmcfarland** [comment]
Alright I have traced this back to its source. I believe what happened was that at one point the database had it as a one-to-one in the same table, while the spec stated that it could have multiple. So I pulled it out into its own table, and just never assigned it a primary key. As a side note the DoSOCS is simply shoving one record onto a package so it isn't currently using the capability.

References:
If you look at the revision history on this file, you'll see when I removed the field from the table and added the new table. https://github.com/spdx-tools/SPDXDatabase/commits/master/SQL/SPDX.sql

Solution: 
` ALTER TABLE package_license_info_from_files ADD Id INT NOT NULL AUTO INCREMENT PRIMARY KEY`

Agreed?
